src/timeline/timeline.tsx

```diff
@@ -94,19 +136,40 @@ export const Timeline = ({
             const variant =
                 _variant || (index === 0 ? "current" : "upcoming-active");
 
+            let pillContent = "";
+            if (statuses) {
+                pillContent = getPillContents(statuses);
+            }
+
+            const contentText = getContentText(content);
+
+            const ariaLabel =
+                getStatus(variant) +
+                " " +
+                title +
+                " " +
+                pillContent +
+                " " +
+                contentText;
+
             return (
-                <TimelineItem key={`timeline-item-${index}`}>
+                <TimelineItem
+                    key={`timeline-item-${index}`}
+                    role="group"
+                    aria-label={ariaLabel}
```

Comment:
`aria-label` is not needed here. it is basically repeating the item content which seems redundant. in reading mode, the screen reader will announce the text twice.